frontend/web/src/assets/lib/jsc3d/jsc3d.webgl.js

```diff
@@ -0,0 +1,1202 @@
+		gl.bufferData(gl.ARRAY_BUFFER, normals, gl.STATIC_DRAW);
+		if(hasTextureCoords) {
+			mesh.compiled.texcoords = gl.createBuffer();
+			gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.texcoords);
+			gl.bufferData(gl.ARRAY_BUFFER, texcoords, gl.STATIC_DRAW);
+		}
+		gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+		mesh.compiled.faceCount = numOfFaces;
+		mesh.compiled.coordCount = coords.length / 3;
+	}
+	else {
+		/*
+		 * Do not need to rebuild, just update normal data.
+		 */
+
+		var isFlat = (mesh.compiled.remderMode == 'flat') || (mesh.compiled.remderMode == 'textureflat');
+		if(isFlat != needFlat) {
+			var normals;
+			if(hasTrianglesOnly) {
+				normals = new Float32Array(9 * numOfFaces);
+				var n0, n1, n2
+				for(var i=0, j=0, faceIndex=0; i<ibuf.length; i+=4, j+=9, faceIndex++) {
+					if(needFlat) {
+						n0 = faceIndex * 3;
+						normals[j    ] = normals[j + 3] = normals[j + 6] = fnbuf[n0    ];
+						normals[j + 1] = normals[j + 4] = normals[j + 7] = fnbuf[n0 + 1];
+						normals[j + 2] = normals[j + 5] = normals[j + 8] = fnbuf[n0 + 2];
+						
+					}
+					else {
+						n0 = nibuf[i    ] * 3;
+						n1 = nibuf[i + 1] * 3;
+						n2 = nibuf[i + 2] * 3;
+						normals[j    ] = nbuf[n0    ];
+						normals[j + 1] = nbuf[n0 + 1];
+						normals[j + 2] = nbuf[n0 + 2];
+						normals[j + 3] = nbuf[n1    ];
+						normals[j + 4] = nbuf[n1 + 1];
+						normals[j + 5] = nbuf[n1 + 2];
+						normals[j + 6] = nbuf[n2    ];
+						normals[j + 7] = nbuf[n2 + 1];
+						normals[j + 8] = nbuf[n2 + 2];
+					}
+				}
+			}
+			else {
+				normals = [];
+				var n0, n1, n2;
+				for(var i=0, j=0; i<numOfFaces; i++) {
+					n0 = nibuf[j++] * 3;
+					n1 = nibuf[j++] * 3;
+					while(ibuf[j] >= 0) {
+						if(needFlat) {
+							n0 = i * 3;
+							normals.push(fnbuf[n0], fnbuf[n0 + 1], fnbuf[n0 + 2], fnbuf[n0], fnbuf[n0 + 1], fnbuf[n0 + 2], fnbuf[n0], fnbuf[n0 + 1], fnbuf[n0 + 2]);
+						}
+						else {
+							n2 = nibuf[j] * 3;
+							normals.push(nbuf[n0], nbuf[n0 + 1], nbuf[n0 + 2], nbuf[n1], nbuf[n1 + 1], nbuf[n1 + 2], nbuf[n2], nbuf[n2 + 1], nbuf[n2 + 2]);
+							n1 = n2;
+						}
+						j++;
+					}
+					j++;
+				}
+				normals = new Float32Array(normals);
+			}
+
+			if(this.isIE11) {
+				// IE11 does not support bufferSubData() for buffer content update. So the normal VBO has to be reallocated for the new data.
+				gl.deleteBuffer(mesh.compiled.normals);
+				mesh.compiled.normals = gl.createBuffer();
+				gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.normals);
+				gl.bufferData(gl.ARRAY_BUFFER, normals, gl.STATIC_DRAW);
+				gl.bindBuffer(gl.ARRAY_BUFFER, null);
+			}
+			else {
+				gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.normals);
+				gl.bufferSubData(gl.ARRAY_BUFFER, 0, normals);
+				gl.bindBuffer(gl.ARRAY_BUFFER, null);
+			}
+		}
+	}
+
+	/*
+	 * Build wireframe if it is not built yet.
+	 */
+	if(renderMode == 'wireframe' && !mesh.compiled.edges) {
+		var edges = makeWireframe(ibuf, vbuf, numOfFaces, hasTrianglesOnly);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "edges" as this name is already used in declaration at line 918. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)